packages/abi-gen/src/utils.ts

```diff
@@ -350,4 +350,28 @@ export const utils = {
             hangingIndent: ' '.repeat(columnsPerIndent),
         });
     },
+    extractTuples(
+        parameter: DataItem,
+        tupleDefinitions: { [pythonTupleName: string]: string }, // output
+        tupleDependencies: Array<[string, string]>, // output
+    ): void {
+        if (parameter.type === 'tuple' || parameter.type === 'tuple[]') {
+            const tupleDataItem = parameter as TupleDataItem; // tslint:disable-line:no-unnecessary-type-assertion
+            // without the above cast (which tslint complains about), tsc says
+            //     Argument of type 'DataItem[] | undefined' is not assignable to parameter of type 'DataItem[]'.
+            //     Type 'undefined' is not assignable to type 'DataItem[]'
+            // when the code below tries to access tupleDataItem.components.
```

Comment:
Can't you check for `undefined` to get rid of the this error?

Reply:
I'm just being stubborn.  I don't want to say `if` anything at all, because I shouldn't have to.  I created `TupleDataItem`, over in `ethereum-types` (in a prior PR), specifically to have a `DataItem` that _definitely does_ have a `components` field, so that I wouldn't have to do all this checking.  Here, `parameter` is a `DataItem`, but when I see that it has `.type === 'tuple'`, I know it's a `TupleDataItem`, so I cast it.  There *is* a difference between `DataItem` and `TupleDataItem` (the latter _requires_ a `components` field, the former merely permits it), so I assert that there is a bug in `tslint` wherein it's complaining that this cast is unnecessary.  And I like the idea of writing comments to explain the use of in-code linter directives, so I did that.